python/lsst/cp/pipe/cpTask.py

```diff
@@ -0,0 +1,594 @@
+
+        Each stage of the processing generates more mask files, so this allows each to be picked up
+        as more and more tests run, and saves having to have clever logic for if some tasks fail.
+        Parameters
+        ----------
+        path : string
+            Path on which to find the mask files
+        ccd : string/int
+            Name/identifier of the CCD
+        Returns
+        -------
+        maskFiles : list/tuple
+            List of mask files, or an empty tuple if none are found
+        """
+        pattern = '*' + str(ccd) + '*mask*'  # the cast to str supports obs_auxTel
+        maskFiles = glob.glob(os.path.join(path, pattern))
+        return maskFiles if len(maskFiles) > 0 else ()  # eotest wants an empty tuple here
+
+    def _cleanupEotest(self, path):
+        """Delete all the medianed files left behind after eotest has run.
+
+        Running eotest generates a lot of interim medianed files, so this just cleans them up.
+        Parameters
+        ----------
+        path : string
+           Path on which to delete all the eotest medianed files.
+        """
+        for filename in glob.glob(os.path.join(path, '*median*')):
+            os.remove(filename)
+
+    def _gainPropSetToDict(self, pSet):
+        """Translator for the persisted gain values.
+
+        eotest wants an {amp: gain} dictionary with integer keys for the amps.
+        When we persist dafBase.propertySets these only take strings for keys,
+        so the provided .toDict() method won't do, so here we provide an extra
+        layer of translation to keep eotest happy.
+
+        Parameters
+        ----------
+        pSet : daf.base.PropertySet
+           PropertySet to be translated to an integer-keyed dictionary
+        """
+        return {int(amp): gain for amp, gain in pSet.toDict().items()}
+
+    def testMethod(self):  # TODO: XXX remove this whole method, just useful for debugging/dev work
+        """TODO: Remove this docstring that only exists to remove flake8 error."""
+        self.log.warn("Test warning message")
+        print(self.config.requireAllEOTests)
+
+    # @pipeBase.timeMethod #xxx re-include this
+    def runEotestDirect(self, butler, run=None):
+        """
+        Generate calibration products using eotest algorithms.
+
+        Generate all calibration products possible using the vanilla eotest implementation,
+        given a butler for a TS8 (raft-test) repo. It can contain multiple runs, but must correspond to
+        only a single raft/RTM.
+
+        - Run all eotest tasks possible, using the butler to gather the data
+        - Write outputs in eotest format
+
+        In order to replicate the canonical eotest analysis, the tasks should be run in a specific order.
+        This is given/defined in the "Steps" section here:
+        http://lsst-camera.slac.stanford.edu/eTraveler/exp/LSST-CAMERA/
+               displayProcess.jsp?processPath=1179
```

Comment:
Don't split the URL.

Reply:
Huh, I could have sworn that I did that because it exceeded the 110 char limit, but apparently it doesn't. Sorry!

Reply:
Even if it exceeds the 110 char limit, you shouldn't split a URL.

Reply:
😱

Reply:
Good to know though, thanks!